plugins/moveToArchive.koplugin/main.lua

```diff
@@ -0,0 +1,163 @@
+local _ = require("gettext")
+
+local MoveToArchive = WidgetContainer:new{
+    name = "move2archive",
+}
+
+local move_to_archive_settings = LuaSettings:open(("%s/%s"):format(DataStorage:getSettingsDir(), "move_to_archive_settings.lua"))
+local archive_dir_config_key = "archive_dir";
+local last_copied_from_config_key = "last_copied_from_dir";
+local archive_dir_path = move_to_archive_settings:readSetting(archive_dir_config_key)
+local last_copied_from_dir = move_to_archive_settings:readSetting(last_copied_from_config_key)
+
+function MoveToArchive:init()
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function MoveToArchive:addToMainMenu(menu_items)
+    menu_items.move_to_archive = {
+        text = _("Move to archive"),
+        sub_item_table = {
+            {
+                text = _("Move current book to archive"),
+                callback = function() self:moveToArchive() end,
+            },
+            {
+                text = _("Copy current book to archive - experimental"),
+                callback = function() self:copyToArchive() end,
+            },
+            {
+                text = _("Go to archive folder"),
+                callback = function()
+                    if not archive_dir_path then
+                        self:showNoArchiveConfirmBox()
+                        return
+                    end
+                    if self.ui.document then
+                        self.ui:onClose()
+                    end
+                    if FileManager.instance then
+                        FileManager.instance:reinit(archive_dir_path)
+                    else
+                        FileManager:showFiles(archive_dir_path)
+                    end
+                end,
+            },
+            {
+                text = _("Go to last copied/moved from folder"),
+                callback = function()
+                    if not last_copied_from_dir then
+                        UIManager:show(InfoMessage:new{
+                            text = _("No previous folder found.")
+                         })
+                        return
+                    end
+                    if self.ui.document then
+                        self.ui:onClose()
+                    end
+                    if FileManager.instance then
+                        FileManager.instance:reinit(last_copied_from_dir)
+                    else
+                        FileManager:showFiles(last_copied_from_dir)
+                    end
+                end,
+            },
+            {
+                text = _("Set archive directory"),
+                keep_menu_open = true,
+                callback =  self.setArchiveDirectory,
+            }
+        },
+    }
+end
+
+function MoveToArchive:moveToArchive()
+    local move_done_text = _("Book moved.\nDo you want to open it from the archive folder?")
+    local doc_old_path = self:commonProcess(true, move_done_text)
+    ReadHistory:removeItemByPath(doc_old_path)
```

Comment:
Better have that done in `self:commonProcess(true...` so all that housekeeping is in a single place).